### PR TITLE
Adds in plasma solar panels, nerfs regular panels

### PR DESCRIPTION
**Pull request by CRITAWAKETS**
## About The Pull Request

Say Hoorah or Oh Gods No, because the solar panel nerf is finally here. Half power for solar production. However, because i like fun, plasma solar panels have been added. These use plasmaglass instead of regular glass, but give double the power production, which is the original values.

## Why It's Good For The Game

You could previously power the entire station just on roundstart solar arrays. Now it's emergency power unless you combine with the turbine, which allows powering the station without using the Supermatter, however that requires fuel and has its own challenges.

The plasma solar panels give both a mid/lategame incentive for engineering to work on, but also allows ghost roles to use old power production values if the new values are just too low.

## Changelog
:cl:
add: Added plasma solar panels, produced by using plasmaglass on a solar assembly. Produces double the power of a regular solar panel.
balance: Solar panels now produce half the power and should not be used to power whole station now.
/:cl:

### CODE REVIEW

File: code/modules/power/solar.dm [hunk 1]
```diff
@@ -185,7 +196,7 @@
 	if(sunfrac <= 0)
 		return
 
-	var/sgen = SOLAR_GEN_RATE * sunfrac
+	var/sgen = SOLAR_GEN_RATE * sunfrac * powermulti
```
Comment (LemonInTheDark) [inline code/modules/power/solar.dm:199]: You don't seem to have halved this define

File: code/modules/power/solar.dm [hunk 2]
```diff
@@ -26,6 +26,8 @@
 	///do we need to call update_solar_exposure() next tick?
 	var/needs_to_update_solar_exposure = TRUE
 	var/obj/effect/overlay/panel
+	var/powermulti = 1 //power multiplier
+	var/pinkpanel = FALSE //controls if it shows a blue or a pink panel sprite
```
Comment (LemonInTheDark) [inline code/modules/power/solar.dm:30]: This needs to be a subtype

File: code/modules/power/solar.dm [hunk 3]
```diff
@@ -69,6 +71,11 @@
 	if(S.glass_type == /obj/item/stack/sheet/rglass) //if the panel is in reinforced glass
 		max_integrity *= 2 								 //this need to be placed here, because panels already on the map don't have an assembly linked to
 		obj_integrity = max_integrity
+	if(S.glass_type == /obj/item/stack/sheet/plasmaglass) //if the panel is made with plasma glass
+		powermulti *= 2 //multiplies power by two
+		pinkpanel = TRUE //makes the solar panel sprite pink
+		max_integrity *= 2
+		obj_integrity = max_integrity
```
Comment (LemonInTheDark) [inline code/modules/power/solar.dm:78]: This should be a part of it

File: code/modules/power/solar.dm [hunk 4]
```diff
@@ -114,8 +121,12 @@
 	var/matrix/turner = matrix()
 	turner.Turn(azimuth_current)
 	panel.transform = turner
-	if(machine_stat & BROKEN)
+	if(machine_stat & BROKEN & !pinkpanel)
 		panel.icon_state = "solar_panel-b"
+	if(machine_stat & BROKEN & pinkpanel)
+		panel.icon_state = "psolar_panel-b"
+	else if (pinkpanel)
+		panel.icon_state = "psolar_panel"
```
Comment (LemonInTheDark) [inline code/modules/power/solar.dm:129]: As should this

**Comment by cacogen** [comment]
Ok, it's not like the SM exploding wasn't used as an excuse to call the shuttle despite the solars being able to provide enough power on their own (at least this is the meme) anyway

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
A: This needs to be a subtype, a bool for sprite/power output/integrity changes is not ok.
B: Don't we want high power to require input from the player past minor setup? I get offstation stuff, but why give players a way to construct them?

**Comment by Cyprex** [comment]
> B: Don't we want high power to require input from the player past minor setup? I get offstation stuff, but why give players a way to construct them?
More importantly constructing them anew or retrofitting plasmaglass onto existing panels is the exact same basic machine construction repeated a lot.
This is just repetitive and unfun, and won't see use the novelty value.

**Comment by Cyprex** [comment]
Fundamentally, solars are just "put machine down, magically get power" with no maintenance or ressource consumption or anything involved(Turbine consumes ressources and is more complex and SM is very complex and has sabotage potential).
It's just very uninteresting. The one thing I could think of that would make things more interesting would be to make power production heavily dependent on placement, possibly randomized per round.

**Comment by 81Denton** [comment]
Make sure to adjust the uhhh oldstation ghost role solar panels for this change

**Comment by Capsandi** [comment]
In theory, solar upkeep is the repair of individual solar panels or portions after asteroids smack into them, the issue is that not enough damage accrues on the huge masses of solar panels that the station starts with. I like the extra usage for plasma glass, but the same effect could be achieved by quickly halving the amount of panels the station starts with, while leaving off station users of solar panels unaffected. This solution also halves the amount of repetitive retrofitting one would have to do to improve the station's power production.

**Comment by Dorsisdwarf** [comment]
This is a pretty good idea, but why have you made it a *variable* in the solar panel rather than a subtype?

**Comment by CRITAWAKETS** [comment]
alright im gonna see if i can slightly unfuck both my own code and the pre-existing solar panel code

**Comment by CRITAWAKETS** [comment]
alright now the plasma solar panels are a subtype and i also did not accidentally get rid of the define halving this time

**Comment by Losenis** [comment]
Would this not also give engineers even less time to work on a SM modification before having to start it up due to solars now only producing half its power?

**Comment by CRITAWAKETS** [comment]
alright so the oldstation and the derelict now have adjusted power production to deal with the new solar production values.

oldstation: replaced 50 glass sheets with 50 plasmaglass sheets.
derelict: replaced top solar array with plasma solar panels, doubled the width of the bottom solar array

**Comment by CRITAWAKETS** [comment]
> Would this not also give engineers even less time to work on a SM modification before having to start it up due to solars now only producing half its power?
 
Yes. That said, a solar panel setup should be able to keep the load off of the roundstart SMES units, giving quite a bit more time than no solars.

**Comment by KathrinBailey** [comment]
Can you even make the assemblies or are they still only accessible via cargo?

File: code/modules/power/solar.dm [hunk 5]
```diff
@@ -251,7 +260,7 @@
 		W.play_tool_sound(src, 75)
 		return TRUE
 
-	if(istype(W, /obj/item/stack/sheet/glass) || istype(W, /obj/item/stack/sheet/rglass))
+	if(istype(W, /obj/item/stack/sheet/glass) || istype(W, /obj/item/stack/sheet/rglass) || istype(W, /obj/item/stack/sheet/plasmaglass))
```
Comment (Fikou) [inline code/modules/power/solar.dm:263]: why do u allow glass, reinforced glass, plasma glass but not reinforced plasma glass

Comment (CRITAWAKETS) [reply code/modules/power/solar.dm:263]: short answer: bloat

long answer: i dont want to have to make an entire new subtype just for "this is plasma glass, but stronger"

**Comment by Yenwodyah** [comment]
I hope you realize that no one is ever going to build these

File: code/modules/power/solar.dm [hunk 6]
```diff
@@ -26,14 +26,23 @@
 	///do we need to call update_solar_exposure() next tick?
 	var/needs_to_update_solar_exposure = TRUE
 	var/obj/effect/overlay/panel
+	var/powermulti = 1 //power multiplier
+	var/solar_panel_sprite = "solar_panel" //used to determine which panel sprite to use
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/power/solar.dm:30]: doc

**Comment by Cyprex** [comment]
Again, with how shitty the construction system is and how many solar panels there are nobody will bother with this. If you had 4x less solars with each one making 4x as much power people might actually be arsed to set these up but with how construction currently works I can't imagine people doing this. I hope you can see how dreadfully repetitive going through all the fucking solars, removing their glass and replacing it with plasmaglass is.
It would be slightly less bad if you could slap existing panels with plasmaglass(can be a do_after) without having to remove the existing glass first.

**Comment by CRITAWAKETS** [comment]
The plan wasnt really to make upgrading the solar panels something you HAVE to do. Just make a better engine or a new array, much more efficient.

**Comment by LemonInTheDark** [comment]
not sure how cobby feels, but I'd still like an answer to this 
`B: Don't we want high power to require input from the player past minor setup? I get offstation stuff, but why give players a way to construct them?`

**Comment by CRITAWAKETS** [comment]
> not sure how cobby feels, but I'd still like an answer to this
> `B: Don't we want high power to require input from the player past minor setup? I get offstation stuff, but why give players a way to construct them?`

With how much time it takes to actually retrofit panels, these are best done for new panels, which require a lot more effort. I want people to be able to make actual powered miniature bases without having to make ridiculous amounts of solar panels.

**Comment by Cyprex** [comment]
> I want people to be able to make actual powered miniature bases without having to make ridiculous amounts of solar panels.

While the idea is cute a pacman(or no powergen and a SMES with filled powercells) seem like less effort. That said, I can't really see where people would actually do a mini-base like that on current /tg/-station

**Comment by cacogen** [comment]
You just do it a couple of screens away from the station and no one will know

**Pull request closed without merging**